qupath-gui-fx/src/main/java/qupath/lib/gui/charts/ScatterPlotDisplay.java

```diff
@@ -0,0 +1,139 @@
+package qupath.lib.gui.charts;
+
+import javafx.geometry.Insets;
+import javafx.scene.control.ComboBox;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+import qupath.lib.gui.QuPathGUI;
+import qupath.lib.gui.dialogs.ParameterPanelFX;
+import qupath.lib.gui.measure.PathTableData;
+import qupath.lib.objects.PathObject;
+import qupath.lib.plugins.parameters.ParameterChangeListener;
+import qupath.lib.plugins.parameters.ParameterList;
+
+/**
+ * A wrapper around {@link ScatterPlotChart} for displaying data about PathObject measurements.
+ */
+public class ScatterPlotDisplay implements ParameterChangeListener {
+
+    private final PathTableData<PathObject> model;
+    private final ComboBox<String> comboNameX = new ComboBox<>();
+    private final ComboBox<String> comboNameY = new ComboBox<>();
+    private final BorderPane pane = new BorderPane();
+    private final ParameterList paramsScatter = new ParameterList()
+            .addIntParameter("nPoints", "Max number of points", 10000, null,  "Maximum number of points to be drawn (>=1)")
+            .addIntParameter("randomSeed", "Random seed", 42, null, "Random seed for point subsampling")
+            .addBooleanParameter("drawGrid", "Draw grid", true, "Whether to draw gridlines on the plot")
+            .addBooleanParameter("drawAxes", "Draw axes", true, "Whether to draw axis ticks on the plot")
+            .addDoubleParameter("pointOpacity", "Point opacity", 1, null, "The opacity of points displayed on the plot.")
```

Comment:
These should have lower and upper bounds, which would result in them being sliders, e.g.
```java
            .addDoubleParameter("pointOpacity", "Point opacity", 1, null, 0, 1, "Set the opacity for individual points (0-1)")
            .addIntParameter("pointSize", "Point size", 4, "px", 1, 10, "Set the size for each individual point");
```
where I set `pointSize` to be an int only because the alternative would give a slider that supports lots of decimal places, which doesn't look great.

Reply:
I didn't particularly like the behaviour of the plot using sliders, but fair enough

Reply:
Sliders, spinners... something bounded. If using `ParameterList` there aren't many options.